pkg/serdes.go

```diff
@@ -37,83 +37,112 @@ func readBodyBuffered(resp *http.Response) ([]byte, error) {
 }
 
 func imageFromBuffer(buffer []byte, transform *ARAwareTransform, aspectRatio float64, encodeImage bool, isMask bool) (*ImagePayload, float64, error) {
-	// Decode the image payload using vips
-	img, err := vips.NewImageFromBuffer(buffer)
+	// Decode the image payload using vips, using bulletproof settings
+	importParams := vips.NewImportParams()
+	importParams.AutoRotate.Set(true)
+	importParams.FailOnError.Set(true)
+	importParams.Page.Set(0)
+	importParams.NumPages.Set(1)
+	importParams.HeifThumbnail.Set(false)
+	importParams.SvgUnlimited.Set(false)
+
+	img, err := vips.LoadImageFromBuffer(buffer, importParams)
 	if err != nil {
-		return nil, -1., err
+		return nil, -1., fmt.Errorf("error loading image: %w", err)
 	}
 
-	err = img.AutoRotate()
-	if err != nil {
-		return nil, -1., err
-	}
-
-	// Optionally crop and resize the image on the fly. Save the aspect ratio in the process for future use
-	originalWidth, originalHeight := img.Width(), img.Height()
-
-	if transform != nil {
-		aspectRatio, err = transform.cropAndResizeToClosestAspectRatio(img, aspectRatio)
-		if err != nil {
-			return nil, -1., err
-		}
-	}
-
-	width, height := img.Width(), img.Height()
-
 	// If the image is 4 channels, we need to drop the alpha channel
 	if img.Bands() == 4 {
 		err = img.Flatten(&vips.Color{R: 255, G: 255, B: 255}) // Flatten with white background
 		if err != nil {
-			fmt.Println("Error flattening image:", err)
-			return nil, -1., err
+			return nil, -1., fmt.Errorf("error flattening image: %w", err)
 		}
 		fmt.Println("Image flattened")
 	}
 
 	// If the image is not a mask but is 1 channel, we want to convert it to 3 channels
 	if (img.Bands() == 1) && !isMask {
-		err = img.ToColorSpace(vips.InterpretationSRGB)
+		if img.Metadata().Format == vips.ImageTypeJPEG {
+			err = img.ToColorSpace(vips.InterpretationSRGB)
+			if err != nil {
+				return nil, -1., fmt.Errorf("error converting to sRGB: %w", err)
+			}
+		} else {
+			// // FIXME: maybe that we could recover these still. By default throws an error, sRGB and PNG not supported
+			return nil, -1., fmt.Errorf("1 channel PNG image not supported")
+		}
+	}
+
+	// If the image is 2 channels, that's gray+alpha and we flatten it
+	if img.Bands() == 2 {
+		err = img.ExtractBand(1, 1)
+		fmt.Println("Gray+alpha image, removing alpha")
 		if err != nil {
-			fmt.Println("Error converting to sRGB:", err)
-			return nil, -1., err
+			return nil, -1., fmt.Errorf("error extracting band: %w", err)
 		}
 	}
 
+	// Optionally crop and resize the image on the fly. Save the aspect ratio in the process for future use
+	originalWidth, originalHeight := img.Width(), img.Height()
+
+	if transform != nil {
```

Comment:
the crux of the stability fix when direct-to-cluster, the subsequent vips call can trigger a sigabrt and I couldn't find any way to catch this case beforehand. 
Flipping this around, this can happen but we recover from it. Guess is that some network transmission was broken and we didn't catch it as we should have